### ISSUE TITLE
remote dir check fails is SE is not in BDII

### ISSUE THREAD

**Issue by ericvaandering** [dmwm/CRAB2#497]
Original Savannah ticket 65253 reported by farinafa on Thu Apr  1 17:08:45 2010.

crab -create checks the remote dir with lcg-ls, no options.
Should instead use lcg-ls -b -Dsrmv2 (like remote stage out does).
Both for consistency with stageout and becasue we also support SE end-points which are not in BDII (at least this test is the only place where those SE's fail).

Stefano

**Comment by ericvaandering** [comment]
Comment by farinafa on Wed Apr 21 08:34:48 2010

Fix is in CVS. 
I introduced a fallback mechanism for those SE that could be not registered in BDII.

This needs some testing even if Federica already checked most of the logic (thanks Federica).

Fabio

**Comment by ericvaandering** [comment]
Comment by belforte on Wed Apr 21 08:40:12 2010

Sorry, that's the wrong approach.
You should simply use same command (lcg-ls -b -Dsrmv2) which is used for stage out. Just add &quot;-b -Dsrmv2&quot;.

Also I do not want that user put by hand the SRM path,
put an incomplete surl, it sails through the check
because lcg-ls adds missing stuff from BDII, and then
stageout fails, defeating the very purpose of the check.

By the way last time we had problems was as simple as CERN having a BDII problem, nothing &quot;structural&quot;, but it is not good to have a dependency in the check that is not there in the stageout.

Bottom line: no access to BDII. Please.
THanks

**Comment by ericvaandering** [comment]
Comment by farinafa on Wed Apr 21 08:47:19 2010

That's exactly what the fix does. 
It adds -b -D if the first call fails. 

In this way, what is done in stage-out (that anyway happens on the WN, i.e. in a completely different environment w.r.t. the UI) is reproduced during remoteCheckDir. But in an &quot;incremental&quot; and conservative way, completely transparent to the users.

Let's see if my approach solves the issue. I'd prefer to face CMSSW_360 bug before refining this one.

Ciao,
   Fabio

**Comment by ericvaandering** [comment]
Comment by belforte on Wed Apr 21 09:00:26 2010

Fabio,
please let's postpone this until 360 etc. is fixed.
THen we need to talk because there is a clear misunderstanding.
stefano

**Comment by ericvaandering** [comment]
Closed by farinafa on Thu Apr 22 10:05:47 2010

**Comment by ericvaandering** [comment]
Comment by farinafa on Thu Apr 22 10:05:47 2010

Forced -b -D srmv2. It will be in 272.

I'm closing the ticket.
